src/OneTimePassword.php

```diff
@@ -113,12 +113,17 @@ public function acceptEntrance()
     {
         $this->update(["status" => "verified"]);
         $this->oneTimePasswordLogs()->where("status", "discarded")->delete();
-        OneTimePassword::where(["status" => "discarded", "user_id" => $this->user->id])->delete();
-        return $this->oneTimePasswordLogs()->where("user_id", $this->user->id)->where("status", "waiting")->update(["status" => "verified"]);
+        OneTimePassword::where(["status" => "discarded", "user_id" => $this->user->getAttribute(config("otp.user_id_field"))])->delete();
+        return $this->oneTimePasswordLogs()->where("user_id", $this->user->getAttribute(config("otp.user_id_field")))->where("status", "waiting")->update(["status" => "verified"]);
     }
 
     public function isExpired()
     {
         return $this->created_at < Carbon::now()->subSeconds(config("otp.otp_timeout"));
     }
+    // check if new OTP needs to send and verify
```

Comment:
This is also redundant, `OneTimePassword::isExpired` method already does that.

Reply:
isExpired() actually check for OTP code timeout, But what if I want that when someone verify OTP it should be only allowed to remain in system for specific amount of time, then he has to verify OTP again. isDiscarded() actually gives this option. In my case I have to make user to verify OTP again after 2 hours, so I set otp_discard_time to 7200 sec, then after 2 hours if he reloads , a new OTP will be sent to his mobile and he has to verify it again. This is why isExpired and isDiscarded are different

Reply:
Ok, I missed that you're checking the `updated_at` column. Does it make sense to use that column to understand that the user has verified the OTP code once?

Reply:
I am checking how much time has passed after user has verified OTP. I am using `updated_at` because it will tell the time when status of that OTP changed to verified. At creation it is waiting, when OTP is correctly entered, status update to verified and I am checking the time when status become verified.

Reply:
Ok, maybe it'll be easier to set the expiration date of this line to the discarding duration, right?

https://github.com/tpaksu/laravel-otp-login/blob/d82bc05184438bd9bb7b2e721832712c6004c3f2/src/LoginMiddleware.php#L98

Reply:
How does this will actually work? It didn't logout user. Also It check for guest, I want it to check for logged in user.

Reply:
The `LoginMiddleware` runs on all **authenticated** requests within the `web` scope, check `LoginMiddleware::willCheck` method. It contains the `Auth::check` check. 

- If the user doesn't have an OTP cookie, and is logged in, the system logs the user out and asks for a re-login, following by the OTP screen.
- If the user does have the OTP cookie, and is logged in, the system continues the request.
- If the user does have the OTP cookie and isn't logged in, after the login, the system skips the OTP screen, if the user ID in the cookie and the logged in user match.
- If the user doesn't have the OTP cookie and isn't logged in, it acts like the first one except the logging out. 
- The cookie doesn't expire until the user explicitly logs out, or their login is expired on the Laravel side (caught a 419 response).
- The OTP code, if isn't verified yet, will expire in `otp.otp_timeout`.
- If the OTP code is expired, a new one will be created.

Reply:
The cookie generated for the user is for 1 year as default. If you want the cookie to get expired after a time of inactivity, you can set the cookie timeout to something different. It refreshes the cookie on each logged in request, so the countdown will start after the user does their last request. This will work like you wanted. 

This is i the function that needs to be parameterized:

https://github.com/tpaksu/laravel-otp-login/blob/master/src/LoginMiddleware.php#L219-L228

Reply:
So cookies expire in 1 year, and user will not havr to regenerate OTP for gis time.?
Also what if user verify his OTP. Then would the cookies expire?
I want that even when a user verify his OTP, it should expire in 2 hours, wether he logout and login again or not. After 2 hours a new OTP would be sent and he has to verify it again.
I actually change time in set cokies to 60 sec, but after 60 sec i was still in the system. On refreshing it should logs me out , discard OTP and send a new one to verify, but it ddnt. Thats why I created isDiscarded() method

Reply:
Ok, let me think more about that feature. At least it needs to be optional. Can you extract the second functionality in a new PR, and keep only the parameterized primary key in this PR? Sorry for the trouble.

Reply:
Sure

Reply:
removed isDiscarded. I think it would be better to create 2nd pull request for is discarded feature after merging 1st one to avoid any posible merge confilcts.